src/js/vic-v2/components/PhotoField.jsx

```diff
@@ -1,9 +1,144 @@
 import React from 'react';
+import Cropper from 'react-cropper';
+import Dropzone from 'react-dropzone';
 
 export default class PhotoField extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      cropResult: null,
+      done: false,
+      zoomValue: 2
+    };
+  }
+
+  onDone = () => {
+    this.setState({ src: null, done: true });
+  }
+
+  onChange = (e) => {
+    e.preventDefault();
+    let files;
+    if (e.dataTransfer) {
+      files = e.dataTransfer.files;
+    } else if (e.target) {
+      files = e.target.files;
+    }
+    const reader = new FileReader();
+    reader.onload = () => {
+      this.setState({ src: reader.result, done: false, cropResult: null });
+    };
+    reader.readAsDataURL(files[0]);
+  }
+
+  onDrop = (files) => {
+    this.setState({ src: files[0].preview });
+  }
+
+  onZoom = (e) => {
+    const newZoomValue = e.detail.ratio;
+    this.setState({ zoomValue: newZoomValue }, () => {
+      this.refs.slider.value = this.state.zoomValue;
+    });
+  }
+
+  moveUp = (e) => {
+    e.preventDefault();
+    this.refs.cropper.move(0, 5);
+  }
+
+  moveDown = (e) => {
+    e.preventDefault();
+    this.refs.cropper.move(0, -5);
+  }
+
+  moveRight = (e) => {
+    e.preventDefault();
+    this.refs.cropper.move(5, 0);
+  }
+
+  moveLeft = (e) => {
+    e.preventDefault();
+    this.refs.cropper.move(-5, 0);
+  }
+
+  zoom = (e) => {
+    e.preventDefault();
+    this.refs.cropper.zoomTo(e.target.value);
+  }
+
+  zoomIn = (e) => {
+    e.preventDefault();
+    this.refs.cropper.zoom(0.1);
+  }
+
+  zoomOut = (e) => {
+    e.preventDefault();
+    this.refs.cropper.zoom(-0.1);
+  }
+
+  cropImage = (e) => {
+    e.preventDefault();
+    const cropResult = this.refs.cropper.getCroppedCanvas().toDataURL();
+    this.setState({ cropResult });
+  }
+
   render() {
+    let instruction = 'Step 1 of 2: Upload a digital photo.';
+    let description = 'Move and resize your photo, so your head and shoulders fit in the square frame below. Click and drag, or use the arrow and magnifying buttons to help.';
+    if (this.state.src) instruction = 'Step 2 of 2: Fit your head and shoulders in the frame';
+    if (this.state.done) description = 'Success! This photo will be printed on your Veteran ID card.';
     return (
-      <div>Imagine a photo upload widget here</div>
+      <div>
+        {!this.state.done && <h4>{instruction}</h4>}
+        {(this.state.src || this.state.done) && <p>{description}</p>}
+        {this.state.done && <img style={{ width: '100%' }} src={this.state.cropResult} alt="cropped"/>}
+        {this.state.src && <div>
+          <Cropper
+            ref="cropper"
+            src={this.state.src}
+            style={{ height: 400, width: '100%' }}
+            aspectRatio={1 / 1}
+            guides={false}
+            viewMode={1}
+            zoom={this.onZoom}
+            crop={this.cropImage}/>
+          <input type="range"
```

Comment:
This slider doesn't seem to work right now.

Reply:
I'm not able to replicate this bug (using the buttons, the slider, and the mouse to zoom, all appear to keep the image zoom and the slider position in sync), could you please indicate what steps are necessary to cause it?

Reply:
If I upload an image and move the slider, it immediately zooms in a bunch. Sliding it back and forth does zoom in and out after that, but it's really zoomed in because of the initial jump and I can't use the slider to set it back the way it was initially.

Reply:
Thanks @jbalboni, I'll investigate this.

Reply:
Thanks @jbalboni, I've adjusted the slider configuration (namely the default value) to address this bug.